module/VuFind/src/VuFind/Hierarchy/TreeDataFormatter/AbstractBaseFactory.php

```diff
@@ -69,7 +69,8 @@ public function __invoke(
         if ($options !== null) {
             throw new \Exception('Unexpected options sent to factory!');
         }
-        $treeDataFormatter = new $requestedName();
+        $config = $container->get(\VuFind\Config\PluginManager::class)->get('HierarchyDefault')->get('HierarchyTree');
```

Comment:
It's been a long time since I've looked at this code, but I can't remember under what circumstances HierarchyDefault gets used, and under what circumstances HierarchyFlat gets used. I think it may be driven by the record driver, in which case it may be impossible to know which configuration is in play at the time this factory executes. If that's the case, maybe moving this setting to the `[Hierarchy]` section of config.ini would be safer.